binding/python3/rigid_body.py

```diff
@@ -10,3 +20,308 @@ def segment_index(model, segment_name):
+            jac[m * 3 : (m + 1) * 3, :] = value.to_array()
+
+        return jac
+
+    def solve(self, method: str = "lm"):
+        """
+        Solve the inverse kinematics by using least_square method from scipy
+
+        Parameters:
+        ----------
+        method: str
+            The method used by least_square to optimize the difference.
+
+            If method = 'lm', the 'trf' method will be used for the first frame, in order to respect the bounds of the model.
+            Then, the 'lm' method will be used for the following frames.
+            If method = 'trf', the 'trf' method will be used for all the frames.
+            If method = 'only_lm', the 'lm' method will be used for all the frames.
+
+            In least_square:
+                -‘trf’ : Trust Region Reflective algorithm, particularly suitable for large sparse problems
+                        with bounds.
+                        Generally robust method.
+                -‘lm’ : Levenberg-Marquardt algorithm as implemented in MINPACK.
+                        Doesn’t handle bounds and sparse Jacobians.
+                        Usually the most efficient method for small unconstrained problems.
+
+        Returns
+        ----------
+        q : np.array
+            generalized coordinates
+        """
+        initial_bounds = (-np.inf, np.inf) if method == "only_lm" else self.bounds
+        inital_method = "lm" if method == "only_lm" else "trf"
+
+        bounds = self.bounds if method == "trf" else (-np.inf, np.inf)
+        method = "lm" if method == "only_lm" else method
+
+        if method != "lm" and method != "trf" and method != "only_lm":
+            raise ValueError('This method is not implemented please use "trf", "lm" or "only_lm" as argument')
+
+        else:
+            for ii in range(0, self.nb_frames):
+                x0 = np.random.random(self.nb_q) * 0.1 if ii == 0 else self.q[:, ii - 1]
+                sol = scipy.optimize.least_squares(
+                    fun=self._marker_diff,
+                    args=(self.xp_markers[:, :, ii], self.idx_to_remove[ii]),
+                    bounds=initial_bounds if ii == 0 else bounds,
+                    jac=self._marker_jac,
+                    x0=x0,
+                    method=inital_method if ii == 0 else method,
+                    xtol=1e-6,
+                    tr_options=dict(disp=False),
+                )
+                self.q[:, ii] = sol.x
+                self.list_sol.append(sol)
+        print("Inverse Kinematics done for all frames")
+        return self.q
+
+    def animate(self):
+        """
+        Animate the result of solve with bioviz.
+        """
+        b = bioviz.Viz(loaded_model=self.biorbd_model, show_muscles=False)
+        b.load_experimental_markers(self.xp_markers)
+        b.load_movement(self.q)
+        b.exec()
+
+    def sol(self):
+        """
+        Create and return a dict which contains the output each optimization.
```

Comment:
"Create and return a dict that contains the output of each optimization."

Reply:
done